packages/components/src/snackbar/index.js

```diff
@@ -34,7 +34,13 @@ function Snackbar( {
 	}, [] );
 
 	const classes = classnames( className, 'components-snackbar' );
-
+	if ( actions && actions.length > 1 ) {
+		// we need to inform developers that snackbar only accepts 1 action
+		// eslint-disable-next-line no-console
+		console.warn( 'Snackbar can only have 1 action, use Notice if your message require many messages' );
+		// return first element only while keeping it inside an array
+		actions = [ actions[ 0 ] ];
+	}
```

Comment:
Another possibility to enforce this is to rename the `actions` prop as `action` in the `Snackbar` component.

Pros:

 - It's clearer
 - No more warnings

Cons:

 - The component API is not consistent with Notice
 - We would dispatch different shapes of `notice` objects in the `@wordpress/notices` module.

cc @aduth thoughts?

Reply:
> Another possibility to enforce this is to rename the `actions` prop as `action` in the `Snackbar` component.
> 
> Pros:
> 
> * It's clearer
> * No more warnings
> 
> Cons:
> 
> * The component API is not consistent with Notice
> * We would dispatch different shapes of `notice` objects in the `@wordpress/notices` module.
> 
> cc @aduth thoughts?

I'm on the fence.

- Multiple properties to serve a similar/same purpose (`action`, `actions`) is non-ideal
- Is it really the case we'll never want to support multiple actions for snackbar notices?

I think warning messaging is appropriate here. We could reevaluate later if it turns to be a pain point.

Reply:
from a design point, it makes sense that we shouldn't support many actions, there is a Notice component for that, this behavior is carried in all major design systems for the same valid reasons.
I should probably ping @kjellr since he implemented the spec and have a better understanding.
I also think warning messages are good enough until we separate the notice component from the snackbar.

[Material Design Snackbar](https://material.io/design/components/snackbars.html#)
[Shopify Polaris Toast](https://polaris.shopify.com/components/feedback-indicators/toast)
[Salesforce Lighting Toast](https://lightningdesignsystem.com/components/toast/)

Reply:
> Is it really the case we'll never want to support multiple actions for snackbar notices?

Yeah, the design team is generally in agreement that we should limit this to one action. Since these are quick, temporary notices, we don't want the user to have to make decisions inside of them. Just a single action is fine, but having the user make a choice could be problematic.